### DISCUSSION TITLE
What purpose do the seemingly redundant length checks in `TakeTwo<T>` serve?

### DISCUSSION THREAD

**Discussion by specious** [pmndrs/zustand#2919] [Q&A]
I'm puzzled by the seemingly redundant `{ length: x | y | z }` checks in the `TakeTwo<T>` type definition (which is like a padded `slice(0, 2)` for list types):

https://github.com/pmndrs/zustand/blob/51c00d22c34de84e6d17c53c32fd142a6f5d6543/src/middleware/devtools.ts#L30-L48

Even the [AI assistant](https://poe.com/s/zN8BBkQBR7c8WQnXrkgX) can't seem to find a case that slips through the direct checks for a `length` property of 0, 1 or 2 explicitly but then still gets caught by the catch-all.

@devanshj, I hate to bother you, but you're likely the best person to answer my question.

**Comment by dai-shi** [comment]
It was like that to support some old TS versions like 3.8. As of now, v5 dropped some old TS versions, so we might be able to simplify it a bit.

**Comment by specious** [comment]
Thank you so much for taking the time to shed some light on this.

If the code had comments explaining this caveat, it wouldn't be so confusing.

**Comment by devanshj** [comment]
See [this comment](https://github.com/pmndrs/zustand/pull/1275#issuecomment-1241933115) and [this PR description](https://github.com/pmndrs/zustand/pull/1348).

**Reply by dai-shi** [reply]
@specious Can you open a PR to add code comments?
